sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/SubnetImpl.java

```diff
@@ -59,9 +61,25 @@ public int networkInterfaceIPConfigurationCount() {
 
     @Override
     public String addressPrefix() {
+        if (!CoreUtils.isNullOrEmpty(this.innerModel().addressPrefixes())) {
```

Comment:
Do we need to have this special logic? Or just return what service returns?

Reply:
Good question, for getters, it's subtle. 
From service perspective, `addressPrefixes` and `addressPrefix` are two separate ideas, and shouldn't be used interchangeably. 
But from SDK perspective, there isn't a need to separate these two, and service behavior is prone to change, leading to potential breaking change. From memory we did something similar before, though I can't remember where...
However, from SDK guideline, if there's a difference between service and SDK, there better be a solid reason. So probably I'll just return service response instead...
@weidongxu-microsoft let us know your thoughts

Reply:
Yeah, maybe we should return what service has in `addressPrefix` (we should already added a warning on javadoc about multiple prefixes).

One reason would be avoiding behavior change from prior SDK version.
Another reason (from me) is this "take first of prefixes" could be misguiding when there are multiple prefixes.

Reply:
Changed to directly return service response.